app/views/shared/vote/_up.html.erb

```diff
@@ -0,0 +1,8 @@
+<%= form_for parent.votes.find_or_initialize_by(user_id: current_user.id) do |f| %>
```

Comment:
is this query done in the presenter?

Reply:
Yes, a method that does a similar query is available there, but I thought I needed to pass the parent to the form partial to get this to work. Are you thinking something like:

`def active_up_button`
    `render_partial("up", { vote: parent.vote })`
  `end`

And then in the partial:
`<%= form_for vote do |f| %>`

??

Reply:
On second glance, if that method in the presenter were updated like so:

```ruby
def vote
  @vote ||= parent.votes.find_or_initialize_by(user_id: h.current_user.id)
end
```

then, yea, the `vote` could be passed to the rendered partial and `form_for vote` could be used here.

Reply:
done. 😈